File: demo/authority/baseline/deno/v1_report.ts
// v1 — the approved version. The Deno twin of ../../v1_approved.lex.
//
// Run as approved:
//   deno run --allow-net=results.demo.internal v1_report.ts
//
// Deno's permission model is the closest thing in wide use to a
// capability grant: per-host network, per-path filesystem, per-variable
// environment, default-deny. Nothing below is a criticism of it.

export function buildReport(runs: number, failures: number): string {
  return `runs=${runs} failures=${failures}`;
}

export function announce(report: string): void {
  console.log(report);
}

export function submit(report: string): Promise<Response> {
  return fetch("https://results.demo.internal/submit", {
    method: "POST",
    body: report,
  });
}

File: demo/authority/baseline/v1_report.py
"""v1 — the approved version. The Python twin of ../v1_approved.lex."""
import requests


def build_report(runs: int, failures: int) -> str:
    return f"runs={runs} failures={failures}"


def announce(report: str) -> None:
    print(report)


def submit(report: str):
    return requests.post("https://results.demo.internal/submit", data=report)
